addons/mail/models/mail_cc_mixin.py

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models, tools
+
+
+class MailCCMixin(models.AbstractModel):
+    _name = 'mail.cc.mixin'
+    _inherit = 'mail.thread'
+    _description = 'Email CC management'
+
+    email_cc = fields.Char('Email cc', help='List of cc from incoming emails.')
```

Comment:
I would ensure content is always sanitized at maximum like 
```
"Name correctly escaped" <sanitized email>
```

I suppose _email_dict purpose is to update CC field only with unknown email adresses and update existing email adresses with new formatted address. This is a good idea. Maybe we could improve a bit your implementation by rebuilding manually what is stored. For example _email_dict could return a dict {'sanitized_email': 'formatted_email'} and we store formatted emails in database, not the raw-one received from incoming emails.

For sanitized emails: use email_normalize in tools so that we always use the same sanitize methodology (basically it lowers everything currently but I think we have plans somewhere in mass mailing to improve it a bit)
For formatted emails: I would manually use a formataddr ("Name with escaping because we could have quotation marks or strange stuff" <sanitized_email>). Indeed default lib formataddr does not put name between brackets and fails with email adresses like "Raoul, the Great <raoul@example.com>".

Reply:
"update existing email addresses with new formatted address."
No actually I want to keep the very first one. If some user exchange mails using same cc with different format, I want to update this field only once. So the email_dict ensure that if we have an existing cc, we dont update it (we update old_dict with new_dict)

Last time we talked (i think it was with you) we discussed about keeping email as raw as possible in database exept if we need to search them (it is not the case here). We can sanitize it before using it, but we cannot find the initial format. We may also want to keep cc if they are in a non sanitize able incorrect format.